### PR TITLE
refactor: :sparkles: address component refactor

### PR DESCRIPTION
**Pull request by KanishkaRajputd**
### Summary by CodeRabbit

Release Notes:
- New Feature: Introduced a new component called `AddressWithHeader` that displays an address with an optional identity badge and additional information.
- New Feature: Added a new enum `EAddressOtherTextType` to represent different types of address text.
- Refactor: Renamed props in the `IdentityBadge` component for clarity and consistency.
- Refactor: Modified prop names in multiple components (`iconSize`, `usernameClassName`, `addressMaxLength`, `showFullAddress`, `truncateUsername`) for consistency.
- Style: Enhanced the UI of the `IdentityBadge` component by adjusting button sizes and styles.
- Bug Fix: Removed the `disableAddressClick` prop from the `Address` component in multiple components.
- Documentation: Updated the props passed to the `Address` component in multiple components.
- Chore: Deduplicated and grouped together files in the summary of changes.

### Summary by CodeRabbit

- Refactor: Updated the `Address` component and related components to improve code clarity and consistency. Renamed several props such as `truncateUsername` to `isTruncateUsername`, `identiconSize` to `iconSize`, and `textClassName` to `usernameClassName`. Removed unnecessary props and added new ones for better control over component behavior.
- Style: Adjusted the styling and layout of various components, including modals and cards, for improved user experience.
- New Feature: Introduced new components `AddressWithHeader` and `InlineAddress` for more flexible address display options.
- Chore: Updated titles in modal components for better clarity, changing "Voted" to "Voted Successfully" and "Delegated" to "Delegated Successfully".

### CODE REVIEW

File: src/ui-components/AddressWithHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	usernameClassName?: string;
+	isTruncateUsername?: boolean;
+	identity?: DeriveAccountRegistration;
+	flags?: DeriveAccountFlags;
+	kiltName?: string;
+	mainDisplay?: string;
+}
+const AddressWithHeader = ({ extensionName, isSubVisible, sub, t2, address, isTruncateUsername, usernameClassName, kiltName, identity, flags, mainDisplay }: Props) => {
+	return (
+		<div className=''>
```
Comment (alphainfinitus) [inline src/ui-components/AddressWithHeader.tsx:24]: `className=''` is redundant

File: src/ui-components/Address.tsx [hunk 2]
```diff
@@ -202,16 +163,21 @@ const Address = ({
...
-	}, [encoded_addr, api, apiReady]);
+	};
 
-	useEffect(() => {
+	const getKiltName = async () => {
+		if (!api || !apiReady) return;
+
+		const web3Name = await getKiltDidName(api, address);
+		setKiltName(web3Name ? `w3n:${web3Name}` : '');
+	};
```
[the remark this answers is not in the document]
Comment (alphainfinitus) [reply src/ui-components/Address.tsx:173]: 😿

File: src/ui-components/Address.tsx [hunk 3]
```diff
@@ -1,111 +1,98 @@
...
+const shortenUsername = (username: string, usernameMaxLength?: number) => {
+	if (username.length > 19) {
+		return shortenAddress(username, usernameMaxLength || 8);
+	}
+	return username;
+};
+
+const FEATURE_RELEASE_DATE = dayjs('2023-06-12').toDate(); // Date from which we are sending custom username flag on web3 sign up.
+
+const Addres = (props: Props) => {
```
Comment (alphainfinitus) [inline src/ui-components/Address.tsx:59]: typo here

**Review by niklabh — APPROVED** [review]

File: src/ui-components/Address.tsx [hunk 4]
```diff
@@ -221,198 +203,136 @@ const Address = ({
...
-	const t2 = extensionName || mainDisplay;
+		(!showFullAddress ? shortenAddress(encodedAddr, addressMaxLength) : encodedAddr) ||
+		shortenUsername(username, usernameMaxLength);
+	const addressSuffix = extensionName || mainDisplay;
+
+	const handleClick = (event: any) => {
+		if (!disableAddressClick) return;
+		event.stopPropagation();
+		event.preventDefault();
+		console.log(disableAddressClick, passedUsername);
```
Comment (alphainfinitus) [inline src/ui-components/Address.tsx:237]: console log

**Review by alphainfinitus — APPROVED** [review]

**Pull request merged**